File: MCRunner/Instruments/Bar.cs
using System;

namespace MCRunner.Instruments
{
    public enum BarType
    {
        Live,
        Historic
    }

    public struct Bar
    {
        public DateTime Time;
        public double High;
        public double Low;
        public double Open;
        public double Close;
        public double Volume;
        public BarType BarType;
    }
}

File: python/mcrunner/instruments/bar.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum, auto

class BarType(Enum):
    Live = auto()
    Historic = auto()

@dataclass
class Bar:
    Time: datetime = None
    High: float = 0.0
    Low: float = 0.0
    Open: float = 0.0
    Close: float = 0.0
    Volume: float = 0.0
    BarType: BarType = BarType.Historic
